src/app/lib/utilities/preprocessor/rules/cpsAssetPage/cpsOnlyOnwardJourneys/index.js

```diff
@@ -0,0 +1,41 @@
+import path from 'ramda/src/path';
+
+const cpsOnlyOnwardJourneys = json => {
+  const groups = path(['relatedContent', 'groups'], json);
+
+  const processedGroups = groups.reduce((acc, group) => {
+    const { type } = group;
+    if (type === 'see-alsos') {
+      const { promos } = group;
+      const onlyCpsPromos = promos.filter(
+        ({ type: promoType }) => promoType === 'cps',
+      );
+
+      const noCpsPromos = onlyCpsPromos.length === 0;
+
+      if (noCpsPromos) {
+        // omit the entire group if no cps promos
+        return acc;
```

Comment:
What value does `acc` return here?

Reply:
So this is doing a `reduce` over the groups array, it makes no assumptions about what other items are in the array.

```
"groups": [
      {
        "type": "see-alsos",
        "promos": [
          {
            "name": "ßíÑß ÏæÌáÇÓ íÔíÏ ÈÞÇÏÉ ÅÓÑÇÆíá",
            "type": "link"
          },
          {
            "name": "ãÇíßá ÏæÌáÇÓ íÓÊÚÏ ááÒæÇÌ ãä ßÇËÑíä ÒíÊÇ",
            "type": "link"
          }
        ]
      }
    ]
```

It filters `promos`, removing all 'non-CPS' promos, after the filter, the above promos array will be empty. When that is the case, we just return the array we have been accumulating, crucially __omiting__ this useless group that has no promos. This serves the purpose of removing the entire 'see-alsos' group, meaning no onward journeys section is rendered.

I wonder if I could simplify this using a filter, a I'll take a look..

Reply:
Just had a look, given that I'm making modifications to the items in `groups` not just filtering them, using `reduce` is correct I think.